### PR TITLE
Convert vsutil to a package.

### PR DESCRIPTION
**Pull request by stuxcrystal**

### CODE REVIEW

**Comment by kageru** [comment]
LGTM, but I don’t really feel qualified since this is about Python packaging. @FichteFoll do you have more experience with that?

**Comment by OrangeChannel** [comment]
Should the version number be bumped to `0.3.0` since there's already a `0.2.0` in the releases tab?

**Comment by stuxcrystal** [comment]
Probably, but I feel it is outside the scope of the PR.

File: setup.py [hunk 1]
```diff
@@ -21,7 +21,10 @@ def run_tests(self):
 setup(
     name='vsutil',
     version='0.1.0',
-    py_modules=['vsutil'],
+    packages=find_packages(exclude=['tests']),
+    package_data={
+        'vsutil': ['py.typed']
+    },
     url='https://encode.moe/vsutil',
```
Comment (FichteFoll) [inline setup.py:28]: This url should be fixed since it currently 404s.

**Comment by heicrd** [comment]
I feel like it'd be cleaner to move `vsutil.py` to `vsutil/vsutil.py` and have `__init__.py` be
```python
"""
VSUtil. A collection of general-purpose VapourSynth functions to be reused in modules and scripts.
"""
from .vsutil import *
__all__ = ['Dither', 'Range', 'depth', 'fallback', 'frame2clip', 'get_depth', 'get_plane_size',
           'get_subsampling', 'get_w', 'get_y', 'insert_clip', 'is_image', 'iterate', 'join', 'plane', 'split']
```
rather than shoving the entire package into `__init__.py`

File: setup.py (hunk 1 above)
Comment (stuxcrystal) [reply setup.py:28]: Let me fix that.

**Comment by stuxcrystal** [comment]
>  Or as an alternative to `__all__`, you can just import the specific attributes from .vsutil

`multiprocessing actually `import *` every module and concatenates the `__all__` of each submodule in the package inside `__init__.py`.

**Comment by stuxcrystal** [comment]
I moved the work on that minimal-import solution to a different branch because stuff kept breaking especially the tests for resolve_enum.

I moved them to the branch stux-minimal-init.

File: setup.py (hunk 1 above)
Comment (stuxcrystal) [reply setup.py:28]: Fixed.

**Comment by FichteFoll** [comment]
> I feel like it'd be cleaner to move `vsutil.py` to `vsutil/vsutil.py` and have `__init__.py` be […]
> rather than shoving the entire package into `__init__.py`

I personally see no need for this, as you still have the entire module/package contents in one file and just add one level of indirection. There is definitely a use case for splitting a package into sub-modules and exporting those on the `package.__init__` level, but this is not happening currently.
Doesn't really matter, either way.

**Comment by OrangeChannel** [comment]
I think it looks a bit odd to a non-Python programmer to have the source code in a strangely-named file in a folder instead of just directly naming the file (or to a VapourSynth func/script writer that is used to just importing from a .py file what they need). But Fichte’s right that for a one-file package like this, leaving stuff in the `__init__` is just easier (and apparently won’t break the tests either).

**Comment by heicrd** [comment]
I think that even as a python programmer it looks a bit odd, as `__init__.py` is intended to be reserved for package initialization.

**Comment by stuxcrystal** [comment]
What about splitting this into two parts:
1. This PR. Makes it a package and declares it as typed.
2. Then we split the library up into smaller logical parts

**Comment by stuxcrystal** [comment]
@kageru 
Since there is no discussion anymore, can you merge it?

**Pull request merged**